File: SOURCE_WEBAPI/VecompSoftware.DocSuiteWeb.Mapper.Model/Messages/Implementation/MessageModelMapper.cs
using VecompSoftware.DocSuiteWeb.Entity.Messages;
using VecompSoftware.DocSuiteWeb.Model.Entities.Messages;

namespace VecompSoftware.DocSuiteWeb.Mapper.Model.Messages
{
    public class MessageModelMapper : BaseModelMapper<Message, MessageModel>, IMessageModelMapper
    {
        #region [ Fields ]
        private readonly IMapperUnitOfWork _mapper;
        #endregion

        #region [ Constructor ]
        public MessageModelMapper(IMapperUnitOfWork mapperUnitOfWork)
        {
            _mapper = mapperUnitOfWork;
        }
        #endregion

        #region [ Methods ]
        public override MessageModel Map(Message entity, MessageModel entityTransformed)
        {
            entityTransformed.IdMessage = entity.EntityId;
            entityTransformed.MessageType = (DocSuiteWeb.Model.Entities.Messages.MessageType)entity.MessageType;
            entityTransformed.RegistrationDate = entity.RegistrationDate;
            entityTransformed.RegistrationUser = entity.RegistrationUser;
            entityTransformed.Status = (DocSuiteWeb.Model.Entities.Messages.MessageStatus)entity.Status;
            entityTransformed.MessageAttachments = _mapper.Repository<IMessageAttachmentModelMapper>().MapCollection(entity.MessageAttachments);
            entityTransformed.MessageEmails = _mapper.Repository<IMessageEmailModelMapper>().MapCollection(entity.MessageEmails);
            entityTransformed.MessageContacts = _mapper.Repository<IMessageContactModelMapper>().MapCollection(entity.MessageContacts);
            return entityTransformed;
        }
        #endregion
    }
}

File: SOURCE_DOCSUITE/VecompSoftware.DocSuiteWeb.Gui/App/Mappers/Messages/MessageModelMapper.ts
import BaseMapper = require('App/Mappers/BaseMapper');
import MessageModel = require("App/Models/Messages/MessageModel");

class MessageModelMapper extends BaseMapper<MessageModel> {
    constructor() {
        super();
    }

    public Map(source: any): MessageModel {
        let toMap: MessageModel = <MessageModel>{};

        if (!source) {
            return null;
        }

        toMap.EntityId = source.EntityId;
        toMap.Status = source.Status;
        toMap.UniqueId = source.UniqueId;
        toMap.RegistrationDate = source.RegistrationDate;
        toMap.RegistrationUser = source.RegistrationUser;
        toMap.MessageContacts = source.MessageContacts;
        toMap.MessageEmails = source.MessageEmails;

        return toMap;
    }
}
export = MessageModelMapper;